src/main/java/org/testng/TestRunner.java

```diff
@@ -539,6 +539,14 @@ private void initMethods() {
             comparator);
   }
 
+  private ITestNGMethod[] buildBeforeClassMethods(Map<Object, ITestNGMethod[]> beforeClassConfig) {
```

Comment:
Should we consider using parallel streams here to speed this up incase there are many objects for a given test class ?